### PR TITLE
Added Unit tests for "logger.js" & "settings.js"

### PR DESCRIPTION
**Pull request by LesikDee**
## Description

Unit tests: 
-  for "logger.js" developed
-  for "settings.js" created

## Type of changes
- New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read [CONTRIBUTING](https://github.com/epam/miew/blob/master/CONTRIBUTING.md) and [CODE_OF_CONDUCT](https://github.com/epam/miew/blob/master/CODE_OF_CONDUCT.md) guides.
- [x] I have followed the code style of this project.
- [x] I have run `yarn run ci`: lint and tests pass locally with my changes.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added necessary documentation / The changes do not need docs update.

### CODE REVIEW

File: src/settings.test.js [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+        testDiffs.set('modes.VD.kSigma', 3);
+        testSettings.set(testDiffs.now);
+        expect(testSettings.now.modes.VD.kSigma).to.be.equal(3);
+      });
+    });
+  });
+
+  describe('.get()', () => {
+    it('gets current parameter', () => {
+      expect(settings.get('modes.VD.kSigma')).to.be.equal(1);
```
Comment (NataliaDSmirnova) [inline src/settings.test.js]: Can we expect a default value to be equal to 1. The default number may change, and the test fails. Should we set kSigma=1 before getting it?

Comment (LesikDee) [reply src/settings.test.js]: Yes, that will be right, thank you! Edited.

File: src/settings.test.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+  describe('.get()', () => {
+    it('gets current parameter', () => {
+      const testSettings = _.cloneDeep(settings);
+      testSettings.now.modes.VD.kSigma = 1;
+      expect(settings.get('modes.VD.kSigma')).to.be.equal(1);
+    });
+  });
+
+  describe('.reset()', () => {
+    it('resets all parameters in default', () => {
```
Comment (paulsmirnov) [inline src/settings.test.js]: Another behavior to verify in a separate statement is notifying about changes during reset.

File: src/settings.test.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+      const testSettings = _.cloneDeep(settings);
+
+      testSettings.set('modes.VD.kSigma', 3);
+
+      testSettings.reset();
+      expect(testSettings).to.be.eql(settings);
```
Comment (paulsmirnov) [inline src/settings.test.js]: What is `eql`? Let's be consistent.

File: src/settings.test.js [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+    });
+  });
+
+  describe('.checkpoint()', () => {
+    it('saved "now" parameters in "old" field', () => {
+      const testSettings = _.cloneDeep(settings);
+      testSettings.now.modes.VD.kSigma = 3;
+      testSettings.checkpoint();
+      expect(testSettings.now.modes.VD.kSigma).to.be.equal(testSettings.old.modes.VD.kSigma);
```
Comment (paulsmirnov) [inline src/settings.test.js]: Again, do not write to `now` directly **anywhere**, use `set()` method. Also:
- maybe we should compare the entire objects (as stated in the test title)?
- maybe we should check for the exact value (to distinguish from `reset()`)?

File: src/settings.test.js [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+        const changeParams = {};
+        changeParams.kSigma = testSettings.now.modes.VD.kSigma;
+        expect(testSettings.changed()[0]).to.equal('modes.VD.kSigma');
+      });
+    });
+  });
+
+  describe('.applyDiffs()', () => {
+    describe('version doesnt match', () => {
```
Comment (paulsmirnov) [inline src/settings.test.js]: First, we usually test the most general, common use (e.g. versions match), and we may even not emphasize version in this case at all. Just "overwrites settings with the diffs". Then, we can add more behaviors: "throws an error if versions do not match", etc.

File: src/utils/logger.js [hunk 6]
```diff
@@ -53,9 +59,7 @@ Object.defineProperty(Logger.prototype, 'level', {
   },
   set(strValue) {
     const lvlVal = logLevels[strValue];
-    if (!_.isNumber(lvlVal)) {
-      throw new Error('Wrong log level specified!');
-    }
+    isNumber(lvlVal);
```
Comment (paulsmirnov) [inline src/utils/logger.js]: Let's choose a better name.

File: src/utils/logger.js [hunk 7]
```diff
@@ -138,7 +140,7 @@ Logger.prototype._message = function (levelVal, message) {
     const output = `miew:${level}: ${message}`;
     if (level === 'error') {
       console.error(output); // NOSONAR
-    } else if (level === 'warning') {
+    } else if (level === 'warn') {
```
Comment (paulsmirnov) [inline src/utils/logger.js]: Do we have a test to avoid such mistakes in the future?

File: src/settings.test.js (hunk 1 above)
Comment (NataliaDSmirnova) [reply src/settings.test.js]: Done.

File: src/settings.test.js [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+import _ from 'lodash';
+import settings from './settings';
+import utils from './utils';
+
+
+describe('/settings', () => {
+  describe('.set()', () => {
+    describe('path is a string', () => {
+      it('sets chosen parameters with current values', () => {
+        const testSettings = _.cloneDeep(settings);
```
[the remark this answers is not in the document]
Comment (NataliaDSmirnova) [reply src/settings.test.js]: Done

File: src/settings.test.js [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+      });
+    });
+
+    describe('path is not a string (is an object)', () => {
+      it('makes diff with settings.now and sets diff', () => {
+        const testSettings = _.cloneDeep(settings);
+
+        const testDiffs = _.cloneDeep(settings);
+        testDiffs.set('modes.VD.kSigma', 3);
+        testSettings.set(testDiffs.now);
```
[the remark this answers is not in the document]
Comment (NataliaDSmirnova) [reply src/settings.test.js]: Done.

File: src/settings.test.js (hunk 3 above)
Comment (NataliaDSmirnova) [reply src/settings.test.js]: 'eql' == 'deep equality'

File: src/settings.test.js (hunk 4 above)
Comment (NataliaDSmirnova) [reply src/settings.test.js]: i've checked the entire objects as checkpoint() preserve the whole structure

File: src/settings.test.js (hunk 5 above)
Comment (NataliaDSmirnova) [reply src/settings.test.js]: I changed the order of the tests

File: src/settings.test.js [hunk 10]
```diff
@@ -0,0 +1,136 @@
+import { expect } from 'chai';
+import _ from 'lodash';
+import settings from './settings';
+import utils from './utils';
+
+
+describe('/settings', () => {
+  describe('.set()', () => {
+    describe('path is a string', () => {
```
[the remark this answers is not in the document]
Comment (NataliaDSmirnova) [reply src/settings.test.js]: Done

File: src/utils/logger.js (hunk 6 above)
Comment (NataliaDSmirnova) [reply src/utils/logger.js]: I've changed to 'verify()'

File: src/utils/logger.js (hunk 7 above)
Comment (NataliaDSmirnova) [reply src/utils/logger.js]: Yes, we have

**Pull request merged**